web/wp-content/themes/dctx-v2/inc/template-tags.php

```diff
@@ -0,0 +1,540 @@
+
+		<a href="<?php echo esc_url( $args['url'] ); ?>" tabindex="-1">
+			<?php if ( $args['image'] ) : ?>
+				<?php echo wp_kses_post( $args['image'] ); ?>
+			<?php else : ?>
+				<img src="<?php echo esc_url( get_stylesheet_directory_uri() ); ?>/assets/images/placeholder.png" class="card-image" loading="lazy" alt="<?php echo sprintf( esc_attr( 'Featured image for %s', 'dctx' ), esc_attr( $args['title'] ) ); ?>">
+			<?php endif; ?>
+		</a>
+
+		<div class="card-section">
+
+		<?php if ( $args['title'] ) : ?>
+			<h3 class="card-title"><a href="<?php echo esc_url( $args['url'] ); ?>"><?php echo esc_html( $args['title'] ); ?></a></h3>
+		<?php endif; ?>
+
+		<?php if ( $args['text'] ) : ?>
+			<p class="card-text"><?php echo esc_html( $args['text'] ); ?></p>
+		<?php endif; ?>
+
+		<?php if ( $args['url'] ) : ?>
+			<a class="button button-card" href="<?php echo esc_url( $args['url'] ); ?>"><?php esc_html_e( 'Read More', 'dctx' ); ?></a>
+		<?php endif; ?>
+
+		</div><!-- .card-section -->
+	</div><!-- .card -->
+	<?php
+}
+
+/**
+ * Display header button.
+ *
+ * @author WDS
+ * @author Corey Collins
+ * @return string
+ */
+function dctx_display_header_button() {
+
+	// Get our button setting.
+	$button_setting = get_theme_mod( 'dctx_header_button' );
+
+	// If we have no button displayed, don't display the markup.
+	if ( 'none' === $button_setting ) {
+		return '';
+	}
+
+	// Grab our button and text values.
+	$button_url  = get_theme_mod( 'dctx_header_button_url' );
+	$button_text = get_theme_mod( 'dctx_header_button_text' );
+	?>
+	<div class="site-header-action">
+		<?php
+		// If we're doing a URL, just make this LOOK like a button but be a link.
+		if ( 'link' === $button_setting && $button_url ) :
+		?>
+			<a href="<?php echo esc_url( $button_url ); ?>" class="button button-link"><?php echo esc_html( $button_text ?: __( 'More Information', 'dctx' ) ); ?></a>
+		<?php else : ?>
+			<button type="button" class="cta-button" aria-expanded="false" aria-label="<?php esc_html_e( 'Search', 'dctx' ); ?>">
+				<?php esc_html_e( 'Search', 'dctx' ); ?>
+			</button>
+			<div class="form-container" aria-hidden="true">
+				<?php get_search_form(); ?>
+			</div><!-- .form-container -->
+		<?php endif; ?>
+	</div><!-- .header-trigger -->
+	<?php
+}
+
+/**
+ * Displays numeric pagination on archive pages.
+ *
+ * @param array $args Array of params to customize output.
+ *
+ * @author WDS
+ * @return void.
+ * @author Corey Collins
+ */
+function dctx_display_numeric_pagination( $args = array() ) {
+
+	// Set defaults.
+	$defaults = array(
+		'prev_text' => '&laquo;',
+		'next_text' => '&raquo;',
+		'mid_size'  => 4,
+	);
+
+	// Parse args.
+	$args = wp_parse_args( $args, $defaults );
+
+	if ( is_null( paginate_links( $args ) ) ) {
+		return;
+	}
+	?>
+
+	<nav class="pagination-container container" aria-label="<?php esc_html_e( 'numeric pagination', 'dctx' ); ?>">
+		<?php echo paginate_links( $args ); // WPCS: XSS OK. ?>
```

Comment:
:warning: **Warning**: Using the WPCS native whitelist comments is deprecated. Please use the PHPCS native &quot;phpcs:ignore Standard.Category.SniffName.ErrorCode&quot; annotations instead. Found: // WPCS: XSS OK.  (*WordPress.Security.EscapeOutput.DeprecatedWhitelistCommentFound*).